include/swift/AST/Identifier.h

```diff
@@ -123,6 +125,21 @@ class Identifier {
   static bool isEditorPlaceholder(StringRef name) {
     return name.startswith("<#");
   }
+    
+  bool canBeArgumentLabel() const {
+    return !empty() && ::swift::canBeArgumentLabel(str());
+  }
+  
+  void getAsArgumentLabel(SmallVectorImpl<char> &buffer) const {
+    bool isReserved = !canBeArgumentLabel();
+    llvm::SmallString<16> newStr;
+    if (isReserved)
+      newStr += '`';
+    newStr += str();
+    if (isReserved)
+      newStr += '`';
+    buffer = newStr;
```

Comment:
Instead of doing this in a temporary SmallString, please work directly in the SmallVectorImpl. I know it's a bit more of a pain, but you should be able to use vector operations to get the data in there.